test/validation/api/event/event.c

```diff
@@ -11,6 +11,197 @@
 #define NUM_EVENTS  100
 #define EVENT_SIZE  100
 #define EVENT_BURST 10
+#define MAX_EVENT_TYPES 5
+
+typedef struct {
+	odp_event_type_t event_types[MAX_EVENT_TYPES];
+	odp_pool_type_t pool_types[MAX_EVENT_TYPES];
+	odp_pool_t pools[MAX_EVENT_TYPES];
+	odp_event_t events[MAX_EVENT_TYPES][EVENT_BURST];
+	int num_types;
+} event_type_ctx_t;
+
+static event_type_ctx_t g_type_ctx;
+
+static int pool_index_of(odp_pool_type_t type)
+{
+	event_type_ctx_t *ctx = &g_type_ctx;
+
+	for (int i = 0; i < ctx->num_types; i++) {
+		if (ctx->pool_types[i] == type)
+			return i;
+	}
+
+	return -1;
+}
+
+static odp_event_t event_types_alloc_event(odp_pool_type_t pool_type)
+{
+	event_type_ctx_t *ctx = &g_type_ctx;
+	int idx = pool_index_of(pool_type);
+	odp_buffer_t buf;
+	odp_packet_t pkt;
+	odp_timeout_t tmo;
+	odp_event_vector_t evv;
+	odp_packet_vector_t pv;
+
+	if (idx < 0)
+		return ODP_EVENT_INVALID;
+
+	switch (pool_type) {
+	case ODP_POOL_BUFFER:
+		buf = odp_buffer_alloc(ctx->pools[idx]);
+		if (buf == ODP_BUFFER_INVALID)
+			return ODP_EVENT_INVALID;
+		return odp_buffer_to_event(buf);
+	case ODP_POOL_PACKET:
+		pkt = odp_packet_alloc(ctx->pools[idx], EVENT_SIZE);
+		if (pkt == ODP_PACKET_INVALID)
+			return ODP_EVENT_INVALID;
+		return odp_packet_to_event(pkt);
+	case ODP_POOL_TIMEOUT:
+		tmo = odp_timeout_alloc(ctx->pools[idx]);
+		if (tmo == ODP_TIMEOUT_INVALID)
+			return ODP_EVENT_INVALID;
+		return odp_timeout_to_event(tmo);
+	case ODP_POOL_EVENT_VECTOR:
+		evv = odp_event_vector_alloc(ctx->pools[idx]);
+		if (evv == ODP_EVENT_VECTOR_INVALID)
+			return ODP_EVENT_INVALID;
+		return odp_event_vector_to_event(evv);
+	case ODP_POOL_VECTOR:
+		pv = odp_packet_vector_alloc(ctx->pools[idx]);
+		if (pv == ODP_PACKET_VECTOR_INVALID)
+			return ODP_EVENT_INVALID;
+		return odp_packet_vector_to_event(pv);
+	default:
+		return ODP_EVENT_INVALID;
+	}
+}
+
+static int event_types_suite_init(void)
+{
+	event_type_ctx_t *ctx = &g_type_ctx;
+	int i, j, n = 0;
+	char name[ODP_POOL_NAME_LEN];
+	odp_pool_capability_t capa;
+
+	memset(&capa, 0, sizeof(capa));
+	if (odp_pool_capability(&capa) != 0)
+		return -1;
+
+	ctx->event_types[n] = ODP_EVENT_BUFFER;
+	ctx->pool_types[n++] = ODP_POOL_BUFFER;
+
+	ctx->event_types[n] = ODP_EVENT_PACKET;
+	ctx->pool_types[n++] = ODP_POOL_PACKET;
+
+	ctx->event_types[n] = ODP_EVENT_TIMEOUT;
+	ctx->pool_types[n++] = ODP_POOL_TIMEOUT;
+
+	if (capa.event_vector.max_num > 0) {
+		ctx->event_types[n] = ODP_EVENT_VECTOR;
+		ctx->pool_types[n++] = ODP_POOL_EVENT_VECTOR;
+	}
+
+	if (capa.vector.max_num > 0) {
+		ctx->event_types[n] = ODP_EVENT_PACKET_VECTOR;
+		ctx->pool_types[n++] = ODP_POOL_VECTOR;
+	}
+
+	ctx->num_types = n;
+
+	for (i = 0; i < ctx->num_types; i++) {
+		odp_pool_param_t prm;
+
+		odp_pool_param_init(&prm);
+
+		prm.type = ctx->pool_types[i];
+		switch (ctx->pool_types[i]) {
+		case ODP_POOL_BUFFER:
+			prm.buf.num = NUM_EVENTS;
+			prm.buf.size = EVENT_SIZE;
+			prm.buf.uarea_size = sizeof(uint64_t);
```

Comment:
E.g. `prm.buf.uarea_size = ODPH_MIN(sizeof(uint64_t), capa.buf.max_uarea_size);` same for all types.